client/components/ReportRerun/index.jsx

```diff
@@ -91,55 +118,111 @@ const ReportRerun = ({ onQueueUpdate }) => {
+        total +
+        run.reportGroups.reduce((sum, group) => sum + group.reportCount, 0),
+      0
+    );
+    if (onTotalRunsAvailable) {
+      onTotalRunsAvailable(totalAvailableRuns);
+    }
+  }, [activeRuns, onTotalRunsAvailable]);
+
+  const hasRerunnableReports = useMemo(() => {
+    return activeRuns.some(
+      run =>
+        run.reportGroups.reduce((sum, group) => sum + group.reportCount, 0) > 0
+    );
+  }, [activeRuns]);
+
   const [createCollectionJobs] = useMutation(CREATE_COLLECTION_JOBS_MUTATION);
 
   const handleRerunClick = async run => {
-    try {
-      await createCollectionJobs({
-        variables: { atVersionId: run.id }
-      });
+    const triggerLoadPromise = triggerLoad(async () => {
+      try {
+        await createCollectionJobs({
+          variables: { atVersionId: run.id }
+        });
 
-      client.query({
-        query: GET_RERUNNABLE_REPORTS_QUERY,
-        variables: { atVersionId: run.id },
-        fetchPolicy: 'network-only'
-      });
+        await client.query({
+          query: GET_RERUNNABLE_REPORTS_QUERY,
+          variables: { atVersionId: run.id },
+          fetchPolicy: 'network-only'
+        });
 
-      client.query({
-        query: GET_UPDATE_EVENTS,
-        variables: { type: 'COLLECTION_JOB' },
-        fetchPolicy: 'network-only'
-      });
-      onQueueUpdate();
+        await refetchEvents();
+
+        onQueueUpdate();
+
+        setTimeout(() => {
+          eventsPanelRef.current?.focus();
+        }, 100);
+
+        announce(
+          `Report generation successfully initiated for ${run.botName} ${run.newVersion}. Monitor events below.`
+        );
+      } catch (error) {
+        console.error('Failed to create collection jobs:', error);
+        announce(
+          `Error initiating report generation for ${run.botName} ${run.newVersion}. Check console for details.`
+        );
+      }
+    }, `Starting automated report generation for ${run.botName} ${run.newVersion}...`);
+
+    try {
+      await triggerLoadPromise;
     } catch (error) {
-      console.error('Error creating collection jobs:', error);
+      if (error) {
+        console.error('Failed to complete the triggerLoad operation:', error);
+        announce(
+          `An error occurred during the report generation process for ${run.botName} ${run.newVersion}. Check console.`
+        );
+      }
     }
   };
 
   const handleRefreshEvents = async () => {
-    await refetchEvents();
+    announce('Refreshing update events...');
+    try {
+      await refetchEvents();
+    } catch (error) {
+      console.error('Failed to refetch events:', error);
+      announce('Error refreshing update events.');
+    }
   };
 
   return (
-    <div className={styles.rerunSection}>
-      {isAdmin && (
-        <RerunDashboard
-          activeRuns={activeRuns}
-          onRerunClick={handleRerunClick}
+    <LoadingStatus message={loadingMessage}>
+      <div className={styles.rerunSection}>
```

Comment:
Can't seem to find `.rerun-section` in case a style has been wrongly chosen.

Reply:
Thank you! [2bd346d](https://github.com/w3c/aria-at-app/pull/1363/commits/2bd346d2cd8b061a282a1c0f9c69df9765046d85)